src/ops/attention.cc

```diff
@@ -844,6 +859,29 @@ bool MultiHeadAttention::measure_operator_cost(
 
 using PCG::Node;
 
+bool operator==(MultiHeadAttentionParams const &lhs,
```

Comment:
I tried to use `std::tie` to remove the verbosity of code like this and potentially allow you to template-automate a bunch of this code. There's an example at https://github.com/lockshaw/FlexFlow/blob/feat/candle_split/include/flexflow/node.h#L40-L42 and https://github.com/lockshaw/FlexFlow/blob/feat/candle_split/src/runtime/node.cc#L15-L17. Not saying you definitely should use it (I'm on the fence), but you can give it a look and lmk what you think

Reply:
Ah, this seems helpful. Will certainly take a look. Found some other issues with the PR while reviewing so will probably get around to this after I fix some of those.

Reply:
Let's leave this out of this PR and we can chat sometime about if we want to add it separately